src/btree/bt_sync.c

```diff
@@ -221,8 +227,46 @@ __sync_file(WT_SESSION_IMPL *session, WT_CACHE_OP syncop)
 				leaf_bytes += page->memory_footprint;
 				++leaf_pages;
 			}
+			++page_counter;
+
 			WT_ERR(__wt_reconcile(
 			    session, walk, NULL, WT_CHECKPOINTING, NULL));
+
+			if (timer && page_counter > 5000) {
+				__wt_epoch(session, &cur_time);
+				time_diff = WT_TIMEDIFF_SEC(cur_time,
+						conn->ckpt_tree_verb_time);
+
+				if (time_diff > 20) {
+
+					int_bytes_svpt =
+					    internal_bytes - int_bytes_svpt;
+					int_page_svpt =
+					    internal_pages - int_page_svpt;
+					leaf_bytes_svpt =
+					    leaf_bytes - leaf_bytes_svpt;
+					leaf_svpt = leaf_pages - leaf_svpt;
+
+					__wt_verbose(session,
+					    WT_VERB_CHECKPOINT,
+					    "Checkpoint has been running and"
+					    " took %" PRIu64 "sec to write: %"
+					    PRIu64" leaf pages (%" PRIu64 "B),"
+					    "%" PRIu64 " internal pages (%"
+					    PRIu64 "B)", time_diff, leaf_svpt,
+					    leaf_bytes_svpt, int_page_svpt,
+					    int_bytes_svpt);
+
+					int_bytes_svpt = internal_bytes;
+					int_page_svpt = internal_pages;
+					leaf_bytes_svpt = leaf_bytes;
+					leaf_svpt = leaf_pages;
+					conn->ckpt_tree_verb_time = cur_time;
+				}
+
+				page_counter = 0;
```

Comment:
I see, this makes sure that page counter gets reset to 0 when it reaches 5000. When timer is false we do not reset it and can possibly overflow, though we wont be using it. Why not increment page counter only when timer is true. Need to put a comment stating what page_counter does.